gum/backend-qnx/gumprocess-qnx.c

```diff
@@ -231,31 +232,55 @@ void
 gum_process_enumerate_modules (GumFoundModuleFunc func,
                                gpointer user_data)
 {
-  GumDlIteratePhdrContext ctx = { func, user_data };
+  gchar * as_path;
+  gint fd, res;
+  gboolean carry_on = TRUE;
+  procfs_mapinfo * mapinfos;
+  gint num_mapinfos;
+  procfs_debuginfo * debuginfo;
+  gint i = 0;
 
-  dl_iterate_phdr (gum_emit_dl_module, &ctx);
-}
+  as_path = g_strdup_printf ("/proc/%d/as", getpid ());
+  fd = open (as_path, O_RDONLY);
+  g_assert (fd != -1);
+  g_free (as_path);
 
-static int
-gum_emit_dl_module (const struct dl_phdr_info * info,
-                    size_t size,
-                    void * data)
-{
-  GumDlIteratePhdrContext * ctx = (GumDlIteratePhdrContext *) data;
-  gboolean carry_on;
-  GumModuleDetails details;
-  GumMemoryRange range;
+  res = devctl (fd, DCMD_PROC_PAGEDATA, 0, 0, &num_mapinfos);
+  g_assert (res != 0);
+
+  mapinfos = g_malloc (sizeof (procfs_mapinfo) * num_mapinfos);
+  debuginfo = g_malloc (sizeof (procfs_debug) + 0x100);
```

Comment:
Something weird with the indentation here or is it just GitHub's UI? (Make sure there's no tabs, and 2 space indentation.)